packages/common/src/api/createApi.ts

```diff
@@ -128,26 +131,45 @@ const buildEndpointHooks = (
       // Retrieve data from cache if lookup args provided
       if (!endpointState[key]) {
         if (
-          !endpoint.options?.idArgKey ||
+          !(endpoint.options?.idArgKey || endpoint.options?.permalinkArgKey) ||
           !endpoint.options?.kind ||
           !endpoint.options?.schemaKey
         )
           return null
-        const { kind, idArgKey, schemaKey } = endpoint.options
-        if (!fetchArgs[idArgKey]) return null
-        const idAsNumber =
-          typeof fetchArgs[idArgKey] === 'number'
+        const { kind, idArgKey, permalinkArgKey, schemaKey } = endpoint.options
+        if (idArgKey && !fetchArgs[idArgKey]) return null
+        if (permalinkArgKey && !fetchArgs[permalinkArgKey]) return null
+
+        const idAsNumber = idArgKey
+          ? typeof fetchArgs[idArgKey] === 'number'
             ? parseInt(fetchArgs[idArgKey])
             : fetchArgs[idArgKey]
-        const initialCachedEntity = cacheSelectors.getEntry(state, {
-          kind,
-          id: idAsNumber
-        })
+          : null
+        const idCachedEntity = idAsNumber
+          ? cacheSelectors.getEntry(state, {
+              kind,
+              id: idAsNumber
+            })
+          : null
+
+        let permalinkCachedEntity = null
+        if (kind === Kind.TRACKS && permalinkArgKey) {
+          permalinkCachedEntity = getTrack(state, {
```

Comment:
cc: @amendelsohn / @dylanjeffers wondering if there's cause to make this more generic, since collections and tracks are fetched by permalink in some cases and users by handle? Maybe a "bring your own" selector?

Reply:
This is a great callout